RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate redux module settings to Typescript

## Proposed changes

Migrate redux module share to typescript

## Issue(s)	


## How to test or reproduce

Log into the app and check if the settings are correct
Change a setting and see if it works correctly

## Screenshots

## Types of changes




## Checklist



## Further comments
